scripts/gen_lib.lua

```diff
@@ -0,0 +1,71 @@
+lpeg = require('lpeg')
```

Comment:
> Maybe that's better, but wouldn't paths.lua then be need to be found by some mechanism? At which point the mechanism can be used to find the paths directly...

This is more about how many data would be fine to pass through that mechanism. E.g. better make user specify one path to build/ then both path to build/ and pass to source.

> I guess I'm too used to languages where including code from the same directory as the code is a thing you just do...

Do not remember using such languages in Neovim, or them being common in my experience:

- Python is directed  by `sys.path` for which you need things like `pip install --editable .` (or $PYTHONPATH). There is `from .foo import smth`, but this is only going to work in imported modules which requires them being in `sys.path`.
- Shell requires hacks with $0, $_ or $BASH_SOURCE.
- lua needs to adjust package.path and package.cpath via e.g. $LUA_INIT (and it is really the worst: most verbose, needs escaping, inconsistent with others which have $\*PATH).
- C needs supplying -I or -isystem argument to compiler. Did not really do modules with Perl, but AFAIK it has `.` in `@INC`, but not something to run from the *codes* directory and not current. `@INC` may be adjusted with `-I` argument.
- CMake includes only from CMAKE_MODULE_PATH, though it is the closest: you may alternatively use `include(cmake/Foo.cmake)` and it would be relative to where CMakeLists.txt is located (including when you use include() from cmake/Foo.cmake, so it is not “including code from the same directory as the code is”).